src/hazelcore/Action_Exp.re

```diff
@@ -1085,16 +1087,18 @@ and syn_perform_line =
     switch (Statics_Pat.syn(ctx, p)) {
     | None => Failed
     | Some((ty_p, _)) =>
-      let def = ZExp.erase(zdef);
-      let def_ctx = Statics_Exp.extend_let_def_ctx(ctx, p, def);
+      let def_ctx = Statics_Exp.extend_let_def_ctx(ctx, p, ZExp.erase(zdef));
       switch (ana_perform(def_ctx, a, (zdef, u_gen), ty_p)) {
       | Failed => Failed
       | CursorEscaped(side) => escape(u_gen, side)
       | Succeeded((new_zdef, u_gen)) =>
-        let new_zline = ZExp.LetLineZE(p, new_zdef);
-        let new_def = ZExp.erase(new_zdef);
-        let body_ctx = Statics_Exp.extend_let_body_ctx(ctx, p, new_def);
-        Succeeded(LineDone((([], new_zline, []), body_ctx, u_gen)));
+        let (new_zdef, u_gen) =
+          Statics_Exp.ana_fix_holes_z(def_ctx, u_gen, new_zdef, ty_p);
```

Comment:
ana_perform should inductively have already fixed the holes so this shouldn't be necessary

Reply:
If I remove this, I start getting impossible syn again in both the annotated and unannotated cases... not sure what's up

Reply:
check the zipper case in ana_perform for case?

Reply:
resolved (possibly)